docs/authentication.md

```diff
@@ -0,0 +1,186 @@
+# Authentication
+
+When your Agents are deployed, to keep things secure, send a token from the client, then verify it on the server. This mirrors the shape used in PartyKit’s auth guide.
+
+## WebSocket authentication
+
+WebSockets are not HTTP, so the handshake is limited.
+
+**What you cannot send**
+
+- Custom headers during the upgrade
+- `Authorization: Bearer ...` on connect
+
+**What works**
+
+- Put a signed, short-lived token in the connection URL as query parameters
+- Verify the token in your server’s connect path
+
+> Tip: never place raw secrets in URLs. Prefer a JWT or a signed token that expires quickly and is scoped to the user or room.
+
+### Same origin
+
+If the client and server share the origin, the browser will send cookies during the WebSocket handshake. Session based auth can work here. Prefer HTTP-only cookies.
+
+### Cross origin
+
+Cookies do not help across origins. Pass credentials in the URL query, then verify on the server.
+
+## Usage examples
+
+### Static authentication
+
+```ts
+import { useAgent } from "agents/react";
+
+function ChatComponent() {
+  const agent = useAgent({
+    agent: "my-agent",
+    query: {
+      token: "demo-token-123",
+      userId: "demo-user"
+    }
+  });
+
+  return <div>Agent: {agent.agent}</div>;
+}
+```
+
+### Async authentication
+
+Build query values right before connect. Use Suspense for async setup.
+
+```ts
+import { useAgent } from "agents/react";
+import { Suspense, useCallback } from "react";
+
+function ChatComponent() {
+  const asyncQuery = useCallback(async () => {
+    const [token, user] = await Promise.all([getAuthToken(), getCurrentUser()]);
+    return {
+      token,
+      userId: user.id,
+      timestamp: Date.now().toString()
+    };
+  }, []);
+
+  const agent = useAgent({
+    agent: "my-agent",
+    query: asyncQuery
+  });
+
+  return <div>Agent: {agent.agent}</div>;
+}
+
+<Suspense fallback={<div>Authenticating...</div>}>
+  <ChatComponent />
+</Suspense>
+```
+
+### JWT refresh pattern
+
+Refresh the token when it expires, then reconnect with fresh credentials.
+
+```ts
+import { useAgent } from "agents/react";
+import { useCallback } from "react";
+
+function useJWTAgent(agentName: string) {
+  const asyncQuery = useCallback(async () => {
+    let token = localStorage.getItem("jwt");
+
+    if (!token || isTokenExpired(token)) {
```

Comment:
how can you tell that a jwt token is expired? 

in practice I think the call will fail, and you have to trigger the async call only then iirc